bricks/presentational/src/loading-step/index.tsx

```diff
@@ -0,0 +1,208 @@
+import React, { useEffect, useMemo } from "react";
+import { EventEmitter, createDecorators } from "@next-core/element";
+import { ReactNextElement, wrapBrick } from "@next-core/react-element";
+import type { lockBodyScroll as _lockBodyScroll } from "@next-bricks/basic/data-providers/lock-body-scroll/lock-body-scroll";
+import type {
+  GeneralIcon,
+  GeneralIconProps,
+} from "@next-bricks/icons/general-icon";
+import "@next-core/theme";
+import styleText from "./styles.shadow.css";
+import { unwrapProvider } from "@next-core/utils/general";
+import classNames from "classnames";
+import { useCurrentTheme } from "@next-core/react-runtime";
+import StepImage from "./step.png";
+import StepDarkImage from "./step-dark.png";
+
+const { defineElement, property, event, method } = createDecorators();
+
+const lockBodyScroll = unwrapProvider<typeof _lockBodyScroll>(
+  "basic.lock-body-scroll"
+);
+const WrappedIcon = wrapBrick<GeneralIcon, GeneralIconProps>("eo-icon");
+
+interface StepItem {
+  title: string;
+  key: string;
+}
+
+/**
+ * 加载步骤框
+ */
+export
+@defineElement("eo-loading-step", {
+  styleTexts: [styleText],
+})
+class EoLoadingStep extends ReactNextElement {
+  /**
+   * 是否可见
+   */
+  @property({
+    type: Boolean,
+  })
+  accessor visible: boolean | undefined;
+
+  /**
+   * 宽度
+   */
+  @property()
+  accessor width: string | undefined;
+
+  /**
+   * 步骤标题
+   */
+  @property()
+  accessor stepTitle: string | undefined;
+
+  /**
+   * 步骤列表
+   */
+  @property({
+    attribute: false,
+  })
+  accessor stepList: StepItem[] | undefined;
+
+  /**
+   * 当前步骤
+   */
+  @property()
+  accessor curStep: string | undefined;
+
+  /**
+   * 打开事件
+   */
+  @event({ type: "open" })
+  accessor #openEvent!: EventEmitter<void>;
+  #handleOpen() {
+    this.#openEvent.emit();
+  }
+
+  /**
+   * 关闭事件
+   */
+  @event({ type: "close" })
+  accessor #closeEvent!: EventEmitter<void>;
+  #handleClose() {
+    this.#closeEvent.emit();
+  }
+
+  /**
+   * 打开
+   */
+  @method()
+  open() {
+    this.visible = true;
+    this.#handleOpen();
+  }
+
+  /**
+   * 关闭
+   */
+  @method({ bound: true })
+  close() {
+    this.visible = false;
+    this.#handleClose();
+  }
+
+  disconnectedCallback(): void {
+    super.disconnectedCallback();
+    lockBodyScroll(this, false);
+  }
+
+  render() {
+    return (
+      <EoLoadingStepComponent
+        curElement={this}
+        visible={this.visible}
+        width={this.width}
+        stepTitle={this.stepTitle}
+        stepList={this.stepList}
+        curStep={this.curStep}
+      />
+    );
+  }
+}
+
+const statusMap = {
+  pending: {
+    icon: <span className={classNames("step-item-icon", "pending-icon")} />,
+  },
+  loading: {
+    icon: (
+      <WrappedIcon
+        lib="antd"
+        theme="outlined"
+        icon="loading"
+        className="step-item-icon"
```

Comment:
loading 的图标不需要转起来吗？

<img width="293" alt="image" src="https://github.com/easyops-cn/next-bricks/assets/2338946/ea34ee22-b5ae-4060-8181-29013decc87e">

Reply:
eo-icon 加个属性 `spinning` 就会转起来了

```suggestion
        className="step-item-icon"
        spinning
```

Reply:
done。覆盖率受到 diagram 影响了，已经 rebase master 了